skglm/tests/test_penalties.py

```diff
@@ -68,5 +68,17 @@ def test_subdiff_diff_block(block_penalty):
     assert_array_less(est.stop_crit_, est.solver.tol)
 
 
+def test_slope():
+    # check that when alphas = [alpha, ..., alpha], SLOPE and L1 solutions are equal
+    alphas = np.full(n_features, alpha)
+    tol = 1e-10
+    est = GeneralizedLinearEstimator(
+        penalty=SLOPE(alphas),
+        solver=FISTA(max_iter=1000, tol=tol, opt_strategy="fixpoint"),
```

Comment:
test fails when ``opt_strategy="subdiff"`` since ``subdiff_distance`` is not implemented for SLOPE and returns ``None``,
However, we don't raise a meaningful exception fo
```shell
    if stop_crit < self.tol:
TypeError: '<' not supported between instances of 'NoneType' and 'float'
```
I have two alternatives to handle this case:
- define a ``subdiff_distance`` method in SLOPE and raise within it an exception
- add ``NotImplemented`` exception in ``BasePenalty``

Reply:
I would go with the first option. It's not so much that we haven't implemented it, but that it does not exist. So raise a ValueError.